File: easy/1614. Maximum Parentheses depth/MaxDepth.java
class MaxDepth {
    public static int maxDepth(String s) {
        int currDepth = 0;
        int maxDepth = 0;
        
        for(int i=0; i<s.length();i++){
            char currChar = s.charAt(i);
            if (currChar == '('){
                currDepth++;
            } else if (currChar == ')'){
                currDepth--;
            }
            
            maxDepth = Math.max(currDepth, maxDepth);
        }
        return maxDepth;
    }

    public static void main(String[] args) {
        System.out.println(maxDepth("(1+(2*3)+((8)/4))+1"));
    }

}

File: easy/1614. Maximum Parentheses depth/max_depth.py
def max_depth(s: str) -> int:
    curr_depth = 0
    max_depth = 0

    for char in s:
        if char == "(":
            curr_depth += 1
        elif char == ")" and curr_depth > 0:
            curr_depth -= 1

        max_depth = max(curr_depth, max_depth)

    return max_depth


print(max_depth("(1+(2*3)+((8)/4))+1"))
